akka-projection-jdbc/src/main/scala/akka/projection/jdbc/JdbcSession.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.jdbc
+
+import java.sql.Connection
+import java.sql.SQLException
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+import akka.annotation.InternalApi
+import akka.japi.function.{ Function => JFunction }
+
+trait JdbcSession {
+  def withConnection[Result](func: JFunction[Connection, Result]): Result
+
+  def commit(): Unit
+  def rollback(): Unit
+  def close(): Unit
+}
+
+object JdbcSession {
+
+  /**
+   * INTERNAL API: run the blocking DB operations on a single thread on the passed ExecutionContext.
+   */
+  @InternalApi
+  private[akka] def withSession[S <: JdbcSession, Result](jdbcSessionFactory: () => S)(func: S => Result)(
```

Comment:
Maybe this should be called `withTransactionalSession`. I assume at-least-once will come later.

Reply:
👍